rust/core-lib/src/line_offset.rs

```diff
@@ -12,15 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::linewrap::Lines;
+#![allow(clippy::range_plus_one)]
+
+use std::ops::Range;
+
 use xi_rope::Rope;
 
+use crate::linewrap::Lines;
+use crate::selection::SelRegion;
+
 /// A trait from which lines and columns in a document can be calculated
 /// into offsets inside a rope an vice versa.
 pub trait LineOffset {
     // use own breaks if present, or text if not (no line wrapping)
 
-    /// Returns the byte offset corresponding to the given visual line.
+    /// Returns the byte offset corresponding to the given line.
```

Comment:
`visual` is an important distinction, here; it indicates a distinction from 'logical' lines, which are just lines that end in a newline character. a 'visual' line may not end in a line break, and may be the result of word wrap.

I guess this whole comment is slightly out of date, having been moved from `view.rs`?

Reply:
`LineOffset` deals with both visual lines and logical lines, so this comment is indeed out of date.

I don't think that one trait coming with two behaviours is ideal in the long run,
but perhaps there is a way out of this.

Reply:
Right, so this was an intentional choice, to the best of my recollection; the idea is that in the absence of line breaks, the hard breaks *are* the visual lines. This means that all edit ops can work with just a buffer, but things 'just work' when you add a view?

It may be that this isn't helpful, but that was the original thinking.